animated-transformer/src/app/animated-transformer/model-evaluator/model-evaluator.component.ts

```diff
@@ -91,4 +160,38 @@ export class ModelEvaluatorComponent {
     this.modelOutput = outputs[0];
   }
 
+  downloadActivations() {
+    const modelData = this.modelData();
+    if (!modelData) {
+      throw new Error('no current trainState');
+    }
+    const currentTask = this.getCurrentTask();
+    const generator = currentTask.makeExamplesGenerator();
+
+    const trainingData = [];
+    const nExamplesToCollect = 1000;
+    for (let i=0; i<nExamplesToCollect; i++) {
+      const example = generator.next();
+      const input = example.value!.input;
+
+      const decoderOutputs = computeDecoder(
+        modelData.tokenRep,
+        modelData.inputPrepFn,
+        modelData.config.transformer.spec,
+        modelData.params,
+        [input]);
+
+      const output = decoderOutputs.layers[0].seqOuput;
+      trainingData.push({
+        'input': input,
+        'mlpOutputs': {
```

Comment:
FYI: If you wanted to save all the computation, I think you could do: 

const data = getData(decoderOutputs);

I guess it depends a bit on where/how much you want to serialse vs edit this code later.